src/formatDate.js

```diff
@@ -9,6 +9,33 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const result = Array(3).fill(0);
+  const fromSeparator = date.match(/[^\w\s]/);
+
+  const toSeparator = toFormat[toFormat.length - 1].match(/[^\w\s]/)
+    ? toFormat[toFormat.length - 1]
+    : fromSeparator;
```

Comment:
The logic for determining `toSeparator` assumes that the last element of `toFormat` is a separator if it matches the regex. This might not always be true, especially if `toFormat` doesn't explicitly include a separator. Consider revising this logic to ensure it correctly identifies the intended separator.